tofu/utils.py

```diff
@@ -149,7 +149,7 @@ def flatten_dict(d, parent_key='', sep=None, deep='ref',
                 elif deep=='copy':
                     v = v.copy(deep='copy')
             new_key = parent_key + sep + k if parent_key else k
-            if isinstance(v, collections.MutableMapping):
+            if isinstance(v, collections.abc.MutableMapping):
```

Comment:
Why going for the abstract class?

Reply:
I got this warning, saw it was possible to use the abstract class without repercussions and ran tests, seems it does work:

`DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working`

Reply:
Ok, good job :-)
Thanks